gravitee-apim-console-webui/src/management/api/api-traffic-v4/analytics/api-analytics-native/api-analytics-native.component.ts

```diff
@@ -14,15 +14,320 @@
+  public leftColumnTransformed$: Observable<ApiAnalyticsWidgetConfig>[];
+  public rightColumnTransformed$: Observable<ApiAnalyticsWidgetConfig>[];
+  public bottomRowTransformed$: Observable<ApiAnalyticsWidgetConfig>[];
+
+  public activeFilters: Signal<ApiAnalyticsNativeFilters> = computed(() => this.mapQueryParamsToFilters(this.activatedRouteQueryParams()));
+
+  apiPlans$ = this.planService
+    .list(this.activatedRoute.snapshot.params.apiId, undefined, ['PUBLISHED', 'DEPRECATED', 'CLOSED'], undefined, 1, 9999)
+    .pipe(
+      map((plans) => plans.data),
+      shareReplay(1),
+    );
+
+  constructor(
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly router: Router,
+    private readonly apiAnalyticsWidgetService: ApiAnalyticsWidgetService,
+  ) {
+    effect(() => {
+      this.apiAnalyticsWidgetService.setUrlParamsData(this.mapQueryParamsToUrlParamsData(this.activatedRouteQueryParams()));
+    });
+  }
+
+  ngOnInit(): void {
+    // Initialize widgets with mock data
+    this.topRowTransformed$ = [
+      of(this.createMockStatsWidget('Downstream Active Connections', 856, '')),
+      of(this.createMockStatsWidget('Upstream Active Connections', 391, '')),
+      of(this.createMockStatsWidget('Messages Produced from Clients', 1234567, '')),
+      of(this.createMockStatsWidget('Messages Produced to Broker', 2345678, '')),
+      of(this.createMockStatsWidget('Messages Consumed from Broker', 987654, '')),
+      of(this.createMockStatsWidget('Messages Consumed to Clients', 876543, '')),
+    ];
+
+    this.leftColumnTransformed$ = [
+      of(this.createMockLineWidget('Message Production Rate', 'Message production rate over time')),
+      of(this.createMockLineWidget('Data Production Rate', 'Data production rate over time')),
+    ];
+
+    this.rightColumnTransformed$ = [
+      of(this.createMockLineWidget('Message Consumption Rate', 'Message consumption rate over time')),
+      of(this.createMockLineWidget('Data Consumption Rate', 'Data consumption rate over time')),
+    ];
+
+    this.bottomRowTransformed$ = [
+      of(this.createMockStackedBarWidget('Authentication Success vs. Failure', 'Authentication success and failure rates over time')),
+    ];
+  }
+
+  onFiltersChange(filters: ApiAnalyticsNativeFilters): void {
+    this.updateQueryParamsFromFilters(filters);
+  }
+
+  onRefreshFilters(): void {
+    this.apiAnalyticsWidgetService.setUrlParamsData(this.mapQueryParamsToUrlParamsData(this.activeFilters()));
+  }
+
+  ngOnDestroy(): void {
+    this.apiAnalyticsWidgetService.clearStatsCache();
+  }
+
+  private updateQueryParamsFromFilters(filters: ApiAnalyticsNativeFilters): void {
+    const queryParams = this.createQueryParamsFromFilters(filters);
+    this.router.navigate([], {
+      queryParams,
+      queryParamsHandling: 'replace',
+    });
+  }
+
+  private createQueryParamsFromFilters(filters: ApiAnalyticsNativeFilters): Record<string, any> {
```

Comment:
what about add a ts type instead `any` here ? to help future refactoring improvement here ?